src/main/java/lc/kra/system/keyboard/GlobalKeyboardHook.java

```diff
@@ -132,19 +138,66 @@ public GlobalKeyboardHook(boolean raw) throws UnsatisfiedLinkError {
 	 * @param event A global key event
 	 */
 	private void keyPressed(GlobalKeyEvent event) {
+		
+		int virtualKeyCode = event.getVirtualKeyCode();
+		
+		// only add if key is not already hold down
+		if(!isKeyHoldDown(event))
+			holdDownKeyCodes.add(virtualKeyCode);
+		
 		for(GlobalKeyListener listener:listeners)
 			listener.keyPressed(event);
 	}
+	
 	/**
 	 * Invoke keyReleased (transition state TS_UP) for all registered listeners
 	 * 
 	 * @param event A global key event
 	 */
 	private void keyReleased(GlobalKeyEvent event) {
+
+		int virtualKeyCode = event.getVirtualKeyCode();
```

Comment:
Same here, just used once so put it down.

Reply:
done